src/test/java/uk/nhs/prm/e2etests/test/RepositoryE2ETest.java

```diff
@@ -712,4 +708,40 @@ void shouldSendUnexpectedMessageFormatsThroughToEhrTransferServiceDeadLetterQueu
             assertThat(ehrTransferServiceParsingDeadLetterQueue.getMessageContaining(message)).isNotNull();
         });
     }
+
+    @Test
+    @Timeout(value = 60, unit = SECONDS)
+    void shouldTransferOut20EHRsWithin1Minute() {
+        String nhsNumberForTestPatient = "9727018076";
+        String previousGpForTestPatient = "M85019";
+        String asidCodeForTestPatient = "200000000149";
+        List<String> outboundConversationIds = new ArrayList<>();
+
+        for (int i = 0; i < 20; i++) {
+            String outboundConversationId = UUID.randomUUID().toString();
+
+            EhrRequestTemplateContext ehrRequestTemplateContext = EhrRequestTemplateContext
+                    .builder()
+                    .outboundConversationId(outboundConversationId.toUpperCase())
+                    .nhsNumber(nhsNumberForTestPatient)
+                    .newGpOdsCode(previousGpForTestPatient)
+                    .asidCode(asidCodeForTestPatient)
+                    .build();
+
+            outboundConversationIds.add(outboundConversationId);
+
+            String ehrRequestMessage = this.templatingService.getTemplatedString(EHR_REQUEST, ehrRequestTemplateContext);
+
+            mhsInboundQueue.sendMessage(ehrRequestMessage, outboundConversationId);
+        }
```

Comment:
Could potentially do something like this `Stream.generate(UUID.randomUUID().toString()).limit(20).forEach(messageId -> {DO LOGIC HERE});` 😄 forgive me if it's syntactically or programmatically incorrect, just from the top of my head!